hs_core/xmlparser.py

```diff
@@ -0,0 +1,61 @@
+import base64
+import os.path
+from StringIO import StringIO
+from xml.sax import xmlreader, saxutils
+from xml.sax.expatreader import ExpatParser
+
+from django.conf import settings
+
+
+XML_CACHE = os.path.join(settings.PROJECT_ROOT, '_xmlcache')
+
+
+class CatalogedExpatParser(ExpatParser):
+    def external_entity_ref(self, context, base, sysid, pubid):
+        if not self._external_ges:
+            return 1
+
+        source = self._ent_handler.resolveEntity(pubid, sysid)
+        source = saxutils.prepare_input_source(source,
+                                               self._source.getSystemId() or
+                                               "")
+
+        # If an entry does not exist in the xml cache, create it.
+        filepath = os.path.join(XML_CACHE, base64.urlsafe_b64encode(pubid))
+        if not os.path.isfile(filepath):
+            with open(filepath, 'w') as f:
+                contents = source.getByteStream().read()
+                source.setByteStream(StringIO(contents))
+                f.write(contents)
```

Comment:
@jbradberry Got it. I am not well-versed on SAX parser at all, but got an idea by just looking at you code. It looks like setting up an environment variable ```PY_SAX_PARSER``` can direct Django to use our own customized ```ExpatParser```. Is there a document link you can share to explain a bit on SAX ExpatParser for xml parsing so I can learn a bit about it? I assume you have tested it out and verified it works with the new customized parser implementation, so a +1 from me.